src/xtgeo/well/_well_oper.py

```diff
@@ -438,3 +429,52 @@ def report_zonation_holes(self, threshold=5):
         clm = ["INDEX", "X_UTME", "Y_UTMN", "Z_TVDSS", "Zone", "Well"]
 
     return pd.DataFrame(wellreport, columns=clm)
+
+
+def mask_shoulderbeds(self, inputlogs, targetlogs, nsamples, strict):
+    """Mask targetlogs around discrete boundaries."""
+    logger.info("Mask shoulderbeds for some logs...")
+
+    # check that inputlogs exists and that they are discrete, and targetlogs
+    useinputs = []
+    for inlog in inputlogs:
+        if inlog not in self._wlogtypes.keys() and strict is True:
+            raise ValueError(f"Input log {inlog} is missing and strict=True")
+        if inlog in self._wlogtypes.keys() and self._wlogtypes[inlog] != "DISC":
+            raise ValueError(f"Input log {inlog} is not of type DISC")
+        if inlog in self._wlogtypes.keys():
+            useinputs.append(inlog)
+
+    usetargets = []
+    for target in targetlogs:
+        if target not in self._wlogtypes.keys() and strict is True:
+            raise ValueError(f"Target log {target} is missing and strict=True")
+        if target in self._wlogtypes.keys():
+            usetargets.append(target)
+
+    if not useinputs or not usetargets:
+        logger.info("Mask shoulderbeds for some logs... nothing done")
+        return False
+
+    # perform filtering
+    def _growfilter(bseries, nleft):
+        if not nleft:
+            return bseries
+        else:
+            return _growfilter(
+                bseries | bseries.shift(-1) | bseries.shift(1), nleft - 1
+            )
+
+    # make a tmp mask log based input logs and use that for mask filterings
+    for inlog in useinputs:
+        series = self._df[inlog]
+        transitions = series.diff().abs() > 0
```

Comment:
Put line 471-474 into a small function that takes a series as input arg, and outputs a boolean series. Then this can be tested effectively with `pytest.mark.parametrize` and a selection of trivial cases, to demonstrate what it does and to prove correctness one corner cases.

Reply:
How to use pytest.mark.parametrize in this case was not obvious to me (my bad...), but I have added a spesific test:
```
def test_mask_shoulderbeds_logs():
    """Test private routine _mask_shoulderbeds_logs."""
    from xtgeo.well._well_oper import _mask_shoulderbeds_logs

    df = pd.DataFrame({"x": [1, 1, 1, 1, 3, 3, 3, 3, 1, 1, 1, 1], "y": range(12)})
    assert df["y"].isna().sum() == 0

    usedf = df.copy()
    _mask_shoulderbeds_logs(usedf, ["x"], ["y"], nsamples=1)
    assert usedf["y"].isna().sum() == 4

    usedf = df.copy()
    _mask_shoulderbeds_logs(usedf, ["x"], ["y"], nsamples=2)
    assert usedf["y"].isna().sum() == 8
```

Reply:
I suggest test code something like this:
```
@pytest.mark.parametrize(
    "logseries, samples, expected",
    [   
        ([], 0, []),
        ([1], 0, [False]),
        ([1], 1, [True]),
        ([1, 2], 0, [False, False]),
        ([1, 2], 1, [True, True]),
        ([1, 2], 10, [True, True]),
    ],  
)
def test_growfilter(logseries, samples, expected):
    from xtgeo.well._well_oper import _mask_shoulderbeds_logs

    assert _mask_shoulderbeds_logs(logseries, samples) == expected
```
The private function needs some minor adjustments for this to work, and then you add some more tuples to the parametrization list (my hunch is that some corner cases will surprise here)